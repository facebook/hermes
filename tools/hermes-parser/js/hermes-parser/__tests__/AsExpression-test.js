/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict-local
 * @format
 */

'use strict';

import {
  parseForSnapshotBabel,
  printForSnapshotESTree,
  parseForSnapshotESTree,
  printForSnapshotBabel,
} from '../__test_utils__/parse';

describe('`as` expression', () => {
  const code = 'x as number;';

  test('ESTree', async () => {
    expect(await printForSnapshotESTree(code)).toBe(code.trim());
    expect(await parseForSnapshotESTree(code)).toMatchSnapshot();
  });

  test('Babel', async () => {
    expect(await parseForSnapshotBabel(code)).toMatchSnapshot();
    expect(await printForSnapshotBabel(code)).toMatchInlineSnapshot(
      `"(x: number);"`,
    );
  });
});
