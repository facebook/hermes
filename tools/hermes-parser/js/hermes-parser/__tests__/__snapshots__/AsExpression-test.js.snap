// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`as\` expression Babel 1`] = `
{
  "body": [
    {
      "expression": {
        "expression": {
          "name": "x",
          "type": "Identifier",
        },
        "type": "TypeCastExpression",
        "typeAnnotation": {
          "type": "TypeAnnotation",
          "typeAnnotation": {
            "type": "NumberTypeAnnotation",
          },
        },
      },
      "type": "ExpressionStatement",
    },
  ],
  "type": "Program",
}
`;

exports[`\`as\` expression ESTree 1`] = `
{
  "body": [
    {
      "directive": null,
      "expression": {
        "expression": {
          "name": "x",
          "optional": false,
          "type": "Identifier",
          "typeAnnotation": null,
        },
        "type": "AsExpression",
        "typeAnnotation": {
          "type": "NumberTypeAnnotation",
        },
      },
      "type": "ExpressionStatement",
    },
  ],
  "type": "Program",
}
`;
